content/blog/build-and-release-hugo-blog-site-using-azure-pipelines.md

```diff
@@ -0,0 +1,208 @@
+
+![Pipeline](../../static/images/build-and-release-hugo-site-using-azure-pipelines/pipeline.png)
+<!-- ![Pipeline](../../../../../images/build-and-release-hugo-site-using-azure-pipelines/pipeline.png) -->
+
+First a build pipeline is triggered whenever I push to master. Using a Hugo build task the static site is generated and published as an artifact of the build. When the build succeeds a release pipeline is triggered which will take the site and push this into the `ronaldbosma.github.io` repository using a Publish to GitHub Pages task.
+
+Although I'm interacting with a git repository when I publish my site, I decided to put this step in a release pipeline and not a build pipeline. I'm releasing my site here. Not building it. This also allows me to add more stages to for example a testing environment or change the target location and type of my site altogether without affecting my build pipeline.
+
+There's another advantage to this solution besides having to do less manual work when publishing new changes to my blog. I can set a publish date in my posts. Posts with a publish date in the future will be ignored when generating my Hugo site. By scheduling my build pipeline to trigger at a regular interval I'm able to schedule posts to be published in the future.
+
+In the next section of this post you'll find a step-by-step explanation of what I did to create this pipeline.
+
+- [Step 1: Prerequisites](#step-1-prerequisites)
+  - [1.1 Install the Hugo extension](#1-1-install-the-hugo-extension)
+  - [1.2 Install the GitHub Pages Publish extension](#1-2-install-the-github-pages-publish-extension)
+  - [1.3 Generate GitHub Personal Access Token](#1-3-generate-github-personal-access-token)
+- [Step 2: Remove submodule](#step-2-remove-submodule)
+- [Step 3: Build Hugo site](#step-3-build-hugo-site)
+  - [3.1 Create build pipeline linked to GitHub](#3-1-create-build-pipeline-linked-to-github)
+  - [3.2 Configure build pipeline](#3-2-configure-build-pipeline)
+- [Step 4: Publish Hugo site](#step-4-publish-hugo-site)
+
+If you already have experience with Azure Pipelines in combination with GitHub repositories you can just setup the [Prerequisites](#step-1-prerequisites) and jump right into configuring the [build pipeline](#3-2-configure-build-pipeline) and [release pipeline](#step-4-publish-hugo-site).
+
+### Step 1: Prerequisites
+
+You'll need an Azure DevOps project that has the 'Pipelines' Azure DevOps service enabled. You can enable this service in your Project settings.
+
+There are also two extension we'll need to install from the Marketplace. These will be installed at the Organization level in Azure DevOps. So be sure you have the proper permissions.
+
+_Both extension use PowerShell. So if you're unable to install these extension. Have a look at the PowerShell scripts in their GitHub repositories. You might find what you need._
```

Comment:
Both extension**s** use PowerShell. So if you're unable to install these extension**s**.